integration/test/validateliveresources/azurecluster_test.go

```diff
@@ -0,0 +1,139 @@
+// +build liveinstallation
+
+package validateliveresources
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capz "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+	"github.com/giantswarm/azure-admission-controller/integration/env"
+	azureclusterpkg "github.com/giantswarm/azure-admission-controller/pkg/azurecluster"
+	"github.com/giantswarm/azure-admission-controller/pkg/filter"
+	"github.com/giantswarm/azure-admission-controller/pkg/generic"
+	"github.com/giantswarm/azure-admission-controller/pkg/mutator"
+)
+
+func TestAzureClusterFiltering(t *testing.T) {
+	ctx := context.Background()
+	logger, _ := micrologger.New(micrologger.Config{})
+	ctrlClient := NewReadOnlyCtrlClient(t)
+
+	var azureClusterList capz.AzureClusterList
+	err := ctrlClient.List(ctx, &azureClusterList)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, azureCluster := range azureClusterList.Items {
+		if !azureCluster.GetDeletionTimestamp().IsZero() {
+			// Skip CRs that are being deleted.
+			continue
+		}
+
+		ownerClusterGetter := func(objectMeta metav1.ObjectMetaAccessor) (capi.Cluster, bool, error) {
+			ownerCluster, ok, err := generic.TryGetOwnerCluster(ctx, ctrlClient, objectMeta)
+			if err != nil {
+				return capi.Cluster{}, false, microerror.Mask(err)
+			}
+
+			return ownerCluster, ok, nil
+		}
+
+		result, err := filter.IsObjectReconciledByLegacyRelease(ctx, logger, ctrlClient, &azureCluster, ownerClusterGetter)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if result == false {
+			objectName := fmt.Sprintf("%s/%s", azureCluster.Namespace, azureCluster.Name)
+			t.Errorf("Expected 'true' (AzureCluster %s is reconciled by legacy release), got 'false' "+
```

Comment:
I believe you are trying to bee too explicit and it becomes even confusing. I think `Expected AzureCluster %s to be reconciled by legacy release` tells what happened, without having to expose the true/false that IMO is not needed when reading test failures.

Reply:
Good point, I was changing this text few times and I got a Frankenstein in the end :grimacing: